specify/specify7
# Handle 404 errors in QueryComboBox

Fixes #5138 

Adding a `strict` param to `rgetPromise` like `fetchResource` does to pass 404 as an expected error and avoiding an uncaught exception
https://github.com/specify/specify7/blob/6a364e00f96d5c81f0853c65b3d12ed0918fc202/specifyweb/frontend/js_src/lib/components/DataModel/resource.ts#L42-L63

- QCBX will now always fetch its resource with `strict=false`. We will no longer see a 404 crash on production when it fetches an invalid resource. Instead, the QCBX value will be cleared and there should be save blocker indicating there is an invalid value
- This should not break existing usages of `rgetPromise` as the strict param is only optional and defaults to true. Older usages will still have the old behavior

Before this PR - 404 crash on invalid resource:

https://github.com/user-attachments/assets/98520da5-5a55-4322-802f-fe6bf7ce6f75

In this PR:


https://github.com/user-attachments/assets/a1be8084-b48a-4e37-be79-fb2a6f3df4ee



### Checklist

      and self-explanatory (or properly documented)

### Testing instructions



- Initialize a QCBX to an invalid resource
- For example:
  - Go to a CO form definition
  - Initialize cataloger to an invalid resource. You can do this with by adding a `initialize` property to the cataloger cell
 ```
initialize="default=/api/specify/agent/2222212211/"
```
  - Your cell could look something like:
 ```
<cell type="field" id="111" name="cataloger" uitype="querycbx" initialize="default=/api/specify/agent/2222212211/"/>
```
- Go to Data Entry > Collection Object
- [ ] Verify app does not crash
- [ ] Verify the QCBX is empty
- [ ] Verify there is a save blocker saying `Invalid value`
